src/internal/pfsdb/json.go

```diff
@@ -0,0 +1,68 @@
+package pfsdb
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+// jsonMap wraps a map[string]string that is stored as JSONB in the database.  While structs can be
+// directly inserted into JSONB columns by the pgx driver, a raw map[string]string cannot be.
+type jsonMap struct {
+	Data map[string]string
+}
+
+var _ sql.Scanner = (*jsonMap)(nil)
```

Comment:
Are we declaring these globals here to fail compilation if jsonMap no longer implements the Scanner interface?

Reply:
Yes.  See [Effective Go](https://go.dev/doc/effective_go):

> One place this situation arises is when it is necessary to guarantee within the package implementing the type that it actually satisfies the interface. If a type—for example, [json.RawMessage](https://go.dev/pkg/encoding/json/#RawMessage)—needs a custom JSON representation, it should implement json.Marshaler, but there are no static conversions that would cause the compiler to verify this automatically. If the type inadvertently fails to satisfy the interface, the JSON encoder will still work, but will not use the custom implementation. To guarantee that the implementation is correct, a global declaration using the blank identifier can be used in the package:

    var _ json.Marshaler = (*RawMessage)(nil)

> In this declaration, the assignment involving a conversion of a *RawMessage to a Marshaler requires that *RawMessage implements Marshaler, and that property will be checked at compile time. Should the json.Marshaler interface change, this package will no longer compile and we will be on notice that it needs to be updated.